vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2321,6 +2367,41 @@ public void setDataProvider(DataProvider<T, ?> dataProvider) {
         return getDataCommunicator().getDataProvider();
     }
 
+    @Override
+    public GridDataView<T> setDataSource(DataProvider<T, ?> dataProvider) {
+        setDataProvider(dataProvider);
+        return getDataView();
+    }
+
+    /**
+     * Getter for getting a generic GridDataView. This should be used only when
+     * neither {@link #getListDataView()} nor #getLazyDataView are applicable
+     * for the underlying dataSource.
+     *
+     * @return DataView instance implementing {@link GridDataView}
+     */
+    @Override
+    public GridDataView<T> getDataView() {
+        return new GridDataViewImpl(dataCommunicator, this);
+    }
+
+    @Override
+    public GridListDataView<T> setDataSource(ListDataProvider<T> dataProvider) {
+        setDataProvider(dataProvider);
+        return getListDataView();
+    }
+
+    @Override
+    public GridListDataView<T> getListDataView() {
+        if (getDataProvider() instanceof ListDataProvider) {
+            return new GridListDataView<>(dataCommunicator, this);
+        }
+        throw new IllegalStateException(String.format(
```

Comment:
`DataProvider` check here is redundant, because it is made inside the constructor.